src/charm.py

```diff
@@ -724,11 +724,19 @@ def _can_start(self, event: StartEvent) -> bool:
             # immediate reboot will make juju re-run the hook
             self.unit.reboot(now=True)
 
+        if not self.mysql_config.custom_config:
+            # empty config mean start never ran, skip next checks
+            return True
+
         # Safeguard if receiving on start after unit initialization
-        if self.unit_initialized:
-            logger.debug("Delegate status update for start handler on initialized unit.")
-            self._on_update_status(None)
-            return False
+        # with retries to allow time for mysqld startup
+        for _ in range(6):
```

Comment:
fix false negative on `self.unit_initialized` running before mysqld has a chance to start on reboot. If mysqld don't start within the 30s, recovery will happen on next update status

Reply:
may be we should use tenacity and raise an assertion error to keep retrying?

Reply:
i prefer to keep it simple since it is something to make recovery quicker